contracts/rust/deployer/src/lib.rs

```diff
@@ -1741,6 +1813,128 @@ mod tests {
+
+        let stake_table_proxy_addr = deploy_stake_table_proxy(
+            &provider,
+            &mut contracts,
+            token_addr,
+            lc_addr,
+            exit_escrow_period,
+            owner,
+        )
+        .await?;
+
+        // Use V2 interface even for V1 contract (V2 ABI is a superset of V1)
+        let stake_table = StakeTableV2::new(stake_table_proxy_addr, &provider);
+
+        let accounts = provider.get_accounts().await?;
+        let test_delegators = [
+            (accounts[0], U256::from(1000)),
+            (accounts[1], U256::from(1000)),
+            (accounts[2], U256::from(10_000)),
+        ];
+
+        let mut rng = StdRng::from_seed([42u8; 32]);
+        let validator_addr = accounts[4];
+        let commission = 0;
+        let bls_key_pair = BLSKeyPair::generate(&mut rng);
+        let state_key_pair = StateKeyPair::generate_from_seed(rng.gen());
+        let bls_vk_sol: G2PointSol = bls_key_pair.ver_key().to_affine().into();
+        let bls_sig_sol: G1PointSol =
+            sign_address_bls(&bls_key_pair, Address::from(*validator_addr)).into();
+        let schnorr_vk_sol: EdOnBN254PointSol = state_key_pair.ver_key().to_affine().into();
+
+        let receipt = stake_table
+            .registerValidator(bls_vk_sol, schnorr_vk_sol, bls_sig_sol.into(), commission)
+            .from(Address::from(*validator_addr))
+            .send()
+            .await?
+            .get_receipt()
+            .await?;
+        assert!(receipt.status());
+
+        let token = EspToken::new(token_addr, &provider);
+
+        for (delegator_addr, amount) in test_delegators.iter() {
+            let _receipt = token
+                .transfer(*delegator_addr, *amount)
+                .from(Address::from(*owner))
+                .send()
+                .await?
+                .get_receipt()
+                .await?;
+
+            let receipt = token
+                .approve(stake_table_proxy_addr, *amount)
+                .from(*delegator_addr)
+                .send()
+                .await?
+                .get_receipt()
+                .await?;
+            assert!(receipt.status());
+
+            let receipt = stake_table
+                .delegate(validator_addr, *amount)
+                .from(*delegator_addr)
+                .send()
+                .await?
+                .get_receipt()
+                .await?;
+            assert!(receipt.status());
+        }
+
+        let fetched_active_stake = fetch_active_stake_for_stake_table_storage_migration_to_v2(
+            &provider,
+            stake_table_proxy_addr,
+        )
+        .await?;
+
+        assert_eq!(
+            fetched_active_stake,
+            test_delegators
+                .iter()
+                .map(|(_, amount)| *amount)
+                .sum::<U256>(),
+        );
+
+        assert!(
+            fetched_active_stake <= token.balanceOf(stake_table_proxy_addr).call().await?,
```

Comment:
Equality check instead? We did not undelegate or exit so the amounts should match exactly.